scipy/stats/_stats.pyx

```diff
@@ -0,0 +1,37 @@
+cimport cython
+from numpy cimport int64_t, intp_t
+import numpy as np
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def _kendall_condis(intp_t[:] x, intp_t[:] y):
+    cdef:
+        intp_t sup = 1 + np.max(y)
+        intp_t[::1] arr = np.zeros(sup, dtype=np.intp)
+        intp_t i = 0, k = 0, size = x.size, idx
+        int64_t con = 0, dis = 0
+
+    with nogil:
+        while i < size:
+            while k < size and x[i] == x[k]:
+                idx = y[k] - 1
+                while idx != 0:
+                    con += arr[idx]
```

Comment:
You can gain another few percents in speed by avoiding computing the number of concordances, which is entirely unnecessary. If you look at the computation of Kendall's τ in #5754 , you'll see that only the number of discordances, joint ties, and one-sided ties are necessary, and you already have them all.